Repaired for fritzOS7

derpicknicker1 opened the pull request:
This node stopped working with fritzOS7.
This will fix the issue. 
(not tested with OS version below 7)

oliverlorenz committed:
cleaned up and using promises now

oliverlorenz committed:
move felix as contributor, and me as author because I'm now the maintainer of this package

oliverlorenz committed:
2.0.0

codmpm committed:
changed icon, color and paletteLabel adapting to https://github.com/bashGroup/node-red-contrib-fritz

codmpm committed:
wording

oliverlorenz committed:
Merge pull request #1 from codmpm/master

Icon, color and paletteLabel

oliverlorenz committed:
2.0.1

oliverlorenz committed:
Update README.md

Added NPM informations

derpicknicker1 committed:
Repaired for fritzOS7

This node stopped woprking with fritzOS7.
This will fix the issue. 
(not tested with OS version below 7)

oliverlorenz commented:
Hey! Sorry, I missed that PR till now. I will check its still working with FritzOS6, before I merge that. Thanks in advance! :smile:

derpicknicker1 commented:
I also have an other fix which uses other data, but i can not verify if it woks for all fritzboxes (i have a 7490 with newest FW).

oliverlorenz commented:
Can you provide this in a second PR?

derpicknicker1 committed:
Change data path

Changed data request path and port to new location

derpicknicker1 committed:
Parsing for new data

- added requirement of 'xml2js'
- now one can pass `online` , `offline` or `all` as msg.payload to
filter results

derpicknicker1 commented:
I added new commits. 
Data looks like this:
```
payload: array[36]
  [0 … 9]
    0: object
        Index: "1"
        IPAddress: "192.0.0.41"
        MACAddress: "B0:XX:XX:XX:XX:XX"
        Active: "0"
        HostName: "SomeDevice"
        InterfaceType: ""
        X_AVM-DE_Port: "0"
        X_AVM-DE_Speed: "0"
        X_AVM-DE_UpdateAvailable: "0"
        X_AVM-DE_UpdateSuccessful: "unknown"
        X_AVM-DE_InfoURL: ""
        X_AVM-DE_Model: ""
        X_AVM-DE_URL: ""
        X_AVM-DE_Guest: "0"
    .
    .
    .
```
You can pass `online`, `offline` or `all` in msg.payload to the node to filter devices.
Example flow:
```
[{"id":"4eaf22df.b692cc","type":"tab","label":"Flow 1","disabled":false,"info":""},{"id":"c5c0cefb.ec732","type":"inject","z":"4eaf22df.b692cc","name":"get_ONLINE_clients","topic":"","payload":"online","payloadType":"str","repeat":"60","crontab":"","once":true,"onceDelay":"3","x":120,"y":60,"wires":[["5f79f04a.f1305"]]},{"id":"5f79f04a.f1305","type":"fritzbox-presence","z":"4eaf22df.b692cc","fritz":"4dfe676e.40b408","x":380,"y":60,"wires":[["3d86ca07.87e646"]]},{"id":"3d86ca07.87e646","type":"debug","z":"4eaf22df.b692cc","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","x":630,"y":60,"wires":[]},{"id":"4dfe676e.40b408","type":"fritzbox-presence-credentials","z":"","username":"USER","password":"PASS","hostname":"fritz.box"}]
```

VincentSC commented:
This works. Just wget the blobs.

The test-code of formal-code has to be changed to:
```
var devs = []

msg.payload.forEach(function(n) {
    if (n.Active>0) {
        devs.push({ ip: n.IPAddress, mac: n.MACAddress, name: n.HostName })
    }
})

return {payload: devs}
```

guidokanschat commented:
What happened to this PR? It seems it never got merged.

einfach113 commented:
Moin Moin..... Egal was ich mache....
ich bekomme immer:
TypeError: Cannot read properties of undefined (reading 'length')
Ich nutze aktuell eine 7590 mit version 7.50.... Oder hat das jemand damit zum laufen bekommen?

novski commented:
PLEASE merge this, or even use the enhancements of [@SHexplorer](https://github.com/oliverlorenz/node-red-contrib-fritzbox-presence/compare/oliverlorenz:master...SHexplorer:master?expand=1) which work without any problems on my 7530 v7.56
THX!
